cl_demo.c

```diff
@@ -3665,6 +3667,19 @@ void CL_Play_f (void)
 			// Close the file on disk now that we have read the file into memory
 			VFS_CLOSE(playbackfile);
 			playbackfile = mmap_file;
+			#ifdef WITH_ZIP
+			if (extracted_from_archive)
+			{
+				char tmp_dir[MAX_OSPATH];
+				if (COM_StripFilename(name, tmp_dir, sizeof(tmp_dir)))
```

Comment:
Please change to return -1 on error and 0 on success and perhaps rename to COM_DirName?

Also, some extra sanity checks ain't wrong to make sure you don't try to remove '/' or 'c:/' etc by mistake